<launch>
    <arg name="name" default="$(env ROS_NAMESPACE)"/>
    <arg name="type" default="Leader"/>
    <arg name="speed" default="0.2"/>

    <arg name="follower_topic" default="cmd_vel_mux/input/navi"/>
    <arg name="other_marker_frame" default="ar_marker_1"/>
    <arg name="start_topic" default="follower/start"/>
    <arg name="done_topic" default="follower/done"/>

    <!-- Tuned values -->
    <arg name="cache_time" default="2.0"/>
    <arg name="target_distance" default="0.155"/>
    <arg name="duration" default="1.0"/>
    <arg name="Krho" default="0.7"/>
    <arg name="Kalpha" default="0.75"/>
    <arg name="Kbeta" default="-0.15"/>

    <arg name="upside_down" default="True"/>

    <node name="move_tb" pkg="move_turtlebot" type="leader_follower.py">
        <param name="name" value="$(arg name)"/>
        <param name="type" value="$(arg type)"/>
        <param name="speed" value="$(arg speed)"/>

        <param name="follower_motor_cmds" value="$(arg follower_topic)"/>
        <param name="name" value="$(env ROS_NAMESPACE)"/>
        <param name="marker_frame_to_follow" value="$(arg other_marker_frame)"/>
        <param name="other_robot_twist_topic" value="/$(env OTHER_ROS_NAMESPACE)/mobile_base/commands/velocity"/>

        <param name="cache_time" value="$(arg cache_time)"/>
        <param name="target_distance" value="$(arg target_distance)"/>
        <param name="Krho" value="$(arg Krho)"/>
        <param name="Kalpha" value="$(arg Kalpha)"/>
        <param name="Kbeta" value="$(arg Kbeta)"/>
        <param name="duration" value="$(arg duration)"/>

        <param name="pin_num" value="7"/>
        <param name="upside_down" value="$(arg upside_down)"/>"

        <param name="start_topic" value="$(arg start_topic)"/>
        <param name="done_topic" value="$(arg done_topic)"/>
    </node>

</launch>